Test_tlsListener fails in go 1.21

### What's wrong?

```
=== RUN   Test_tlsListener
    tls_test.go:67: 
        	Error Trace:	/build/grafana-agent/src/agent-0.35.4/pkg/server/tls_test.go:67
        	Error:      	"remote error: tls: certificate required" does not contain "tls: bad certificate"
        	Test:       	Test_tlsListener
--- FAIL: Test_tlsListener (0.00s)
```

### Steps to reproduce

`go test -v ./pkg/server`

### System information

go version go1.21.0 linux/amd64

### Software version

agent-0.35.4

### Configuration

_No response_

### Logs

_No response_

--- a/.drone/drone.yml
+++ b/.drone/drone.yml
@@ -126,7 +126,7 @@ steps:
 - commands:
   - apt-get update -y && apt-get install -y libsystemd-dev
   - make lint
-  image: grafana/agent-build-image:0.29.0
+  image: grafana/agent-build-image:0.30.0
   name: Lint
 trigger:
   event:
@@ -144,7 +144,7 @@ steps:
   - ERR_MSG="Dashboard definitions are out of date. Please run 'make generate-dashboards'
     and commit changes!"
   - if [ ! -z "$(git status --porcelain)" ]; then echo $ERR_MSG >&2; exit 1; fi
-  image: grafana/agent-build-image:0.29.0
+  image: grafana/agent-build-image:0.30.0
   name: Regenerate dashboards
 trigger:
   event:
@@ -162,7 +162,7 @@ steps:
   - ERR_MSG="Custom Resource Definitions are out of date. Please run 'make generate-crds'
     and commit changes!"
   - if [ ! -z "$(git status --porcelain)" ]; then echo $ERR_MSG >&2; exit 1; fi
-  image: grafana/agent-build-image:0.29.0
+  image: grafana/agent-build-image:0.30.0
   name: Regenerate crds
 trigger:
   event:
@@ -180,7 +180,7 @@ steps:
   - ERR_MSG="The environment manifests are out of date. Please run 'make generate-manifests'
     and commit changes!"
   - if [ ! -z "$(git status --porcelain)" ]; then echo $ERR_MSG >&2; exit 1; fi
-  image: grafana/agent-build-image:0.29.0
+  image: grafana/agent-build-image:0.30.0
   name: Regenerate environment manifests
 trigger:
   event:
@@ -195,7 +195,7 @@ platform:
 steps:
 - commands:
   - make GO_TAGS="nodocker" test
-  image: grafana/agent-build-image:0.29.0
+  image: grafana/agent-build-image:0.30.0
   name: Run Go tests
 trigger:
   event:
@@ -210,7 +210,7 @@ platform:
 steps:
 - commands:
   - K8S_USE_DOCKER_NETWORK=1 make test
-  image: grafana/agent-build-image:0.29.0
+  image: grafana/agent-build-image:0.30.0
   name: Run Go tests
   volumes:
   - name: docker
@@ -233,7 +233,7 @@ platform:
 steps:
 - commands:
   - go test -tags="nodocker,nonetwork" ./...
-  image: grafana/agent-build-image:0.29.0-windows
+  image: grafana/agent-build-image:0.30.0-windows
   name: Run Go tests
 trigger:
   ref:
@@ -248,7 +248,7 @@ platform:
 steps:
 - commands:
   - make agent-image
-  image: grafana/agent-build-image:0.29.0
+  image: grafana/agent-build-image:0.30.0
   name: Build container
   volumes:
   - name: docker
@@ -273,7 +273,7 @@ platform:
 steps:
 - commands:
   - make agentctl-image
-  image: grafana/agent-build-image:0.29.0
+  image: grafana/agent-build-image:0.30.0
   name: Build container
   volumes:
   - name: docker
@@ -298,7 +298,7 @@ platform:
 steps:
 - commands:
   - make operator-image
-  image: grafana/agent-build-image:0.29.0
+  image: grafana/agent-build-image:0.30.0
   name: Build container
   volumes:
   - name: docker
@@ -324,7 +324,7 @@ platform:
 steps:
 - commands:
   - '& "C:/Program Files/git/bin/bash.exe" ./tools/ci/docker-containers-windows agent'
-  image: grafana/agent-build-image:0.29.0-windows
+  image: grafana/agent-build-image:0.30.0-windows
   name: Build container
   volumes:
   - name: docker
@@ -350,7 +350,7 @@ platform:
 steps:
 - commands:
   - '& "C:/Program Files/git/bin/bash.exe" ./tools/ci/docker-containers-windows agentctl'
-  image: grafana/agent-build-image:0.29.0-windows
+  image: grafana/agent-build-image:0.30.0-windows
   name: Build container
   volumes:
   - name: docker
@@ -377,7 +377,7 @@ steps:
   - make generate-ui
   - GO_TAGS="builtinassets promtail_journal_enabled" GOOS=linux GOARCH=amd64 GOARM=
     make agent
-  image: grafana/agent-build-image:0.29.0
+  image: grafana/agent-build-image:0.30.0
   name: Build
 trigger:
   event:
@@ -394,7 +394,7 @@ steps:
   - make generate-ui
   - GO_TAGS="builtinassets promtail_journal_enabled" GOOS=linux GOARCH=arm64 GOARM=
     make agent
-  image: grafana/agent-build-image:0.29.0
+  image: grafana/agent-build-image:0.30.0
   name: Build
 trigger:
   event:
@@ -411,7 +411,7 @@ steps:
   - make generate-ui
   - GO_TAGS="builtinassets promtail_journal_enabled" GOOS=linux GOARCH=ppc64le GOARM=
     make agent
-  image: grafana/agent-build-image:0.29.0
+  image: grafana/agent-build-image:0.30.0
   name: Build
 trigger:
   event:
@@ -428,7 +428,7 @@ steps:
   - make generate-ui
   - GO_TAGS="builtinassets promtail_journal_enabled" GOOS=linux GOARCH=s390x GOARM=
     make agent
-  image: grafana/agent-build-image:0.29.0
+  image: grafana/agent-build-image:0.30.0
   name: Build
 trigger:
   event:
@@ -444,7 +444,7 @@ steps:
 - commands:
   - make generate-ui
   - GO_TAGS="builtinassets" GOOS=darwin GOARCH=amd64 GOARM= make agent
-  image: grafana/agent-build-image:0.29.0
+  image: grafana/agent-build-image:0.30.0
   name: Build
 trigger:
   event:
@@ -460,7 +460,7 @@ steps:
 - commands:
   - make generate-ui
   - GO_TAGS="builtinassets" GOOS=darwin GOARCH=arm64 GOARM= make agent
-  image: grafana/agent-build-image:0.29.0
+  image: grafana/agent-build-image:0.30.0
   name: Build
 trigger:
   event:
@@ -476,7 +476,7 @@ steps:
 - commands:
   - make generate-ui
   - GO_TAGS="builtinassets" GOOS=windows GOARCH=amd64 GOARM= make agent
-  image: grafana/agent-build-image:0.29.0
+  image: grafana/agent-build-image:0.30.0
   name: Build
 trigger:
   event:
@@ -492,7 +492,7 @@ steps:
 - commands:
   - make generate-ui
   - GO_TAGS="builtinassets" GOOS=freebsd GOARCH=amd64 GOARM= make agent
-  image: grafana/agent-build-image:0.29.0
+  image: grafana/agent-build-image:0.30.0
   name: Build
 trigger:
   event:
@@ -509,7 +509,7 @@ steps:
   - make generate-ui
   - GO_TAGS="builtinassets promtail_journal_enabled" GOOS=linux GOARCH=amd64 GOARM=
     make agent-flow
-  image: grafana/agent-build-image:0.29.0
+  image: grafana/agent-build-image:0.30.0
   name: Build
 trigger:
   event:
@@ -526,7 +526,7 @@ steps:
   - make generate-ui
   - GO_TAGS="builtinassets promtail_journal_enabled" GOOS=linux GOARCH=arm64 GOARM=
     make agent-flow
-  image: grafana/agent-build-image:0.29.0
+  image: grafana/agent-build-image:0.30.0
   name: Build
 trigger:
   event:
@@ -543,7 +543,7 @@ steps:
   - make generate-ui
   - GO_TAGS="builtinassets promtail_journal_enabled" GOOS=linux GOARCH=ppc64le GOARM=
     make agent-flow
-  image: grafana/agent-build-image:0.29.0
+  image: grafana/agent-build-image:0.30.0
   name: Build
 trigger:
   event:
@@ -560,7 +560,7 @@ steps:
   - make generate-ui
   - GO_TAGS="builtinassets promtail_journal_enabled" GOOS=linux GOARCH=s390x GOARM=
     make agent-flow
-  image: grafana/agent-build-image:0.29.0
+  image: grafana/agent-build-image:0.30.0
   name: Build
 trigger:
   event:
@@ -576,7 +576,7 @@ steps:
 - commands:
   - make generate-ui
   - GO_TAGS="builtinassets" GOOS=darwin GOARCH=amd64 GOARM= make agent-flow
-  image: grafana/agent-build-image:0.29.0
+  image: grafana/agent-build-image:0.30.0
   name: Build
 trigger:
   event:
@@ -592,7 +592,7 @@ steps:
 - commands:
   - make generate-ui
   - GO_TAGS="builtinassets" GOOS=darwin GOARCH=arm64 GOARM= make agent-flow
-  image: grafana/agent-build-image:0.29.0
+  image: grafana/agent-build-image:0.30.0
   name: Build
 trigger:
   event:
@@ -608,7 +608,7 @@ steps:
 - commands:
   - make generate-ui
   - GO_TAGS="builtinassets" GOOS=windows GOARCH=amd64 GOARM= make agent-flow
-  image: grafana/agent-build-image:0.29.0
+  image: grafana/agent-build-image:0.30.0
   name: Build
 trigger:
   event:
@@ -624,7 +624,7 @@ steps:
 - commands:
   - make generate-ui
   - GO_TAGS="builtinassets" GOOS=freebsd GOARCH=amd64 GOARM= make agent-flow
-  image: grafana/agent-build-image:0.29.0
+  image: grafana/agent-build-image:0.30.0
   name: Build
 trigger:
   event:
@@ -641,7 +641,7 @@ steps:
   - make generate-ui
   - GO_TAGS="builtinassets promtail_journal_enabled" GOOS=linux GOARCH=amd64 GOARM=
     make agentctl
-  image: grafana/agent-build-image:0.29.0
+  image: grafana/agent-build-image:0.30.0
   name: Build
 trigger:
   event:
@@ -658,7 +658,7 @@ steps:
   - make generate-ui
   - GO_TAGS="builtinassets promtail_journal_enabled" GOOS=linux GOARCH=arm64 GOARM=
     make agentctl
-  image: grafana/agent-build-image:0.29.0
+  image: grafana/agent-build-image:0.30.0
   name: Build
 trigger:
   event:
@@ -675,7 +675,7 @@ steps:
   - make generate-ui
   - GO_TAGS="builtinassets promtail_journal_enabled" GOOS=linux GOARCH=ppc64le GOARM=
     make agentctl
-  image: grafana/agent-build-image:0.29.0
+  image: grafana/agent-build-image:0.30.0
   name: Build
 trigger:
   event:
@@ -692,7 +692,7 @@ steps:
   - make generate-ui
   - GO_TAGS="builtinassets promtail_journal_enabled" GOOS=linux GOARCH=s390x GOARM=
     make agentctl
-  image: grafana/agent-build-image:0.29.0
+  image: grafana/agent-build-image:0.30.0
   name: Build
 trigger:
   event:
@@ -708,7 +708,7 @@ steps:
 - commands:
   - make generate-ui
   - GO_TAGS="builtinassets" GOOS=darwin GOARCH=amd64 GOARM= make agentctl
-  image: grafana/agent-build-image:0.29.0
+  image: grafana/agent-build-image:0.30.0
   name: Build
 trigger:
   event:
@@ -724,7 +724,7 @@ steps:
 - commands:
   - make generate-ui
   - GO_TAGS="builtinassets" GOOS=darwin GOARCH=arm64 GOARM= make agentctl
-  image: grafana/agent-build-image:0.29.0
+  image: grafana/agent-build-image:0.30.0
   name: Build
 trigger:
   event:
@@ -740,7 +740,7 @@ steps:
 - commands:
   - make generate-ui
   - GO_TAGS="builtinassets" GOOS=windows GOARCH=amd64 GOARM= make agentctl
-  image: grafana/agent-build-image:0.29.0
+  image: grafana/agent-build-image:0.30.0
   name: Build
 trigger:
   event:
@@ -756,7 +756,7 @@ steps:
 - commands:
   - make generate-ui
   - GO_TAGS="builtinassets" GOOS=freebsd GOARCH=amd64 GOARM= make agentctl
-  image: grafana/agent-build-image:0.29.0
+  image: grafana/agent-build-image:0.30.0
   name: Build
 trigger:
   event:
@@ -773,7 +773,7 @@ steps:
   - make generate-ui
   - GO_TAGS="builtinassets promtail_journal_enabled" GOOS=linux GOARCH=amd64 GOARM=
     make operator
-  image: grafana/agent-build-image:0.29.0
+  image: grafana/agent-build-image:0.30.0
   name: Build
 trigger:
   event:
@@ -790,7 +790,7 @@ steps:
   - make generate-ui
   - GO_TAGS="builtinassets promtail_journal_enabled" GOOS=linux GOARCH=arm64 GOARM=
     make operator
-  image: grafana/agent-build-image:0.29.0
+  image: grafana/agent-build-image:0.30.0
   name: Build
 trigger:
   event:
@@ -807,7 +807,7 @@ steps:
   - make generate-ui
   - GO_TAGS="builtinassets promtail_journal_enabled" GOOS=linux GOARCH=ppc64le GOARM=
     make operator
-  image: grafana/agent-build-image:0.29.0
+  image: grafana/agent-build-image:0.30.0
   name: Build
 trigger:
   event:
@@ -824,7 +824,7 @@ steps:
   - make generate-ui
   - GO_TAGS="builtinassets promtail_journal_enabled" GOOS=linux GOARCH=s390x GOARM=
     make operator
-  image: grafana/agent-build-image:0.29.0
+  image: grafana/agent-build-image:0.30.0
   name: Build
 trigger:
   event:
@@ -840,7 +840,7 @@ steps:
 - commands:
   - make generate-ui
   - GO_TAGS="builtinassets" GOOS=darwin GOARCH=amd64 GOARM= make operator
-  image: grafana/agent-build-image:0.29.0
+  image: grafana/agent-build-image:0.30.0
   name: Build
 trigger:
   event:
@@ -856,7 +856,7 @@ steps:
 - commands:
   - make generate-ui
   - GO_TAGS="builtinassets" GOOS=darwin GOARCH=arm64 GOARM= make operator
-  image: grafana/agent-build-image:0.29.0
+  image: grafana/agent-build-image:0.30.0
   name: Build
 trigger:
   event:
@@ -872,7 +872,7 @@ steps:
 - commands:
   - make generate-ui
   - GO_TAGS="builtinassets" GOOS=windows GOARCH=amd64 GOARM= make operator
-  image: grafana/agent-build-image:0.29.0
+  image: grafana/agent-build-image:0.30.0
   name: Build
 trigger:
   event:
@@ -888,7 +888,7 @@ steps:
 - commands:
   - make generate-ui
   - GO_TAGS="builtinassets" GOOS=freebsd GOARCH=amd64 GOARM= make operator
-  image: grafana/agent-build-image:0.29.0
+  image: grafana/agent-build-image:0.30.0
   name: Build
 trigger:
   event:
@@ -905,7 +905,7 @@ steps:
   - make generate-ui
   - GO_TAGS="builtinassets promtail_journal_enabled" GOOS=linux GOARCH=amd64 GOARM=
     GOEXPERIMENT=boringcrypto make agent-boringcrypto
-  image: grafana/agent-build-image:0.29.0
+  image: grafana/agent-build-image:0.30.0
   name: Build
 trigger:
   event:
@@ -922,7 +922,7 @@ steps:
   - make generate-ui
   - GO_TAGS="builtinassets promtail_journal_enabled" GOOS=linux GOARCH=arm64 GOARM=
     GOEXPERIMENT=boringcrypto make agent-boringcrypto
-  image: grafana/agent-build-image:0.29.0
+  image: grafana/agent-build-image:0.30.0
   name: Build
 trigger:
   event:
@@ -938,7 +938,7 @@ steps:
 - commands:
   - docker run --rm --privileged multiarch/qemu-user-static --reset -p yes
   failure: ignore
-  image: grafana/agent-build-image:0.29.0
+  image: grafana/agent-build-image:0.30.0
   name: Configure QEMU
   volumes:
   - name: docker
@@ -958,7 +958,7 @@ steps:
       from_secret: docker_password
     GCR_CREDS:
       from_secret: gcr_admin
-  image: grafana/agent-build-image:0.29.0
+  image: grafana/agent-build-image:0.30.0
   name: Publish container
   volumes:
   - name: docker
@@ -982,7 +982,7 @@ steps:
 - commands:
   - docker run --rm --privileged multiarch/qemu-user-static --reset -p yes
   failure: ignore
-  image: grafana/agent-build-image:0.29.0
+  image: grafana/agent-build-image:0.30.0
   name: Configure QEMU
   volumes:
   - name: docker
@@ -1002,7 +1002,7 @@ steps:
       from_secret: docker_password
     GCR_CREDS:
       from_secret: gcr_admin
-  image: grafana/agent-build-image:0.29.0
+  image: grafana/agent-build-image:0.30.0
   name: Publish container
   volumes:
   - name: docker
@@ -1026,7 +1026,7 @@ steps:
 - commands:
   - docker run --rm --privileged multiarch/qemu-user-static --reset -p yes
   failure: ignore
-  image: grafana/agent-build-image:0.29.0
+  image: grafana/agent-build-image:0.30.0
   name: Configure QEMU
   volumes:
   - name: docker
@@ -1046,7 +1046,7 @@ steps:
       from_secret: docker_password
     GCR_CREDS:
       from_secret: gcr_admin
-  image: grafana/agent-build-image:0.29.0
+  image: grafana/agent-build-image:0.30.0
   name: Publish container
   volumes:
   - name: docker
@@ -1070,7 +1070,7 @@ steps:
 - commands:
   - docker run --rm --privileged multiarch/qemu-user-static --reset -p yes
   failure: ignore
-  image: grafana/agent-build-image:0.29.0
+  image: grafana/agent-build-image:0.30.0
   name: Configure QEMU
   volumes:
   - name: docker
@@ -1090,7 +1090,7 @@ steps:
       from_secret: docker_password
     GCR_CREDS:
       from_secret: gcr_admin
-  image: grafana/agent-build-image:0.29.0
+  image: grafana/agent-build-image:0.30.0
   name: Publish container
   volumes:
   - name: docker
@@ -1114,7 +1114,7 @@ steps:
 - commands:
   - docker run --rm --privileged multiarch/qemu-user-static --reset -p yes
   failure: ignore
-  image: grafana/agent-build-image:0.29.0
+  image: grafana/agent-build-image:0.30.0
   name: Configure QEMU
   volumes:
   - name: docker
@@ -1134,7 +1134,7 @@ steps:
       from_secret: docker_password
     GCR_CREDS:
       from_secret: gcr_admin
-  image: grafana/agent-build-image:0.29.0
+  image: grafana/agent-build-image:0.30.0
   name: Publish container
   volumes:
   - name: docker
@@ -1158,7 +1158,7 @@ steps:
 - commands:
   - docker run --rm --privileged multiarch/qemu-user-static --reset -p yes
   failure: ignore
-  image: grafana/agent-build-image:0.29.0
+  image: grafana/agent-build-image:0.30.0
   name: Configure QEMU
   volumes:
   - name: docker
@@ -1178,7 +1178,7 @@ steps:
       from_secret: docker_password
     GCR_CREDS:
       from_secret: gcr_admin
-  image: grafana/agent-build-image:0.29.0
+  image: grafana/agent-build-image:0.30.0
   name: Publish container
   volumes:
   - name: docker
@@ -1207,7 +1207,7 @@ steps:
       from_secret: docker_login
     DOCKER_PASSWORD:
       from_secret: docker_password
-  image: grafana/agent-build-image:0.29.0-windows
+  image: grafana/agent-build-image:0.30.0-windows
   name: Build containers
   volumes:
   - name: docker
@@ -1236,7 +1236,7 @@ steps:
       from_secret: docker_login
     DOCKER_PASSWORD:
       from_secret: docker_password
-  image: grafana/agent-build-image:0.29.0-windows
+  image: grafana/agent-build-image:0.30.0-windows
   name: Build containers
   volumes:
   - name: docker
@@ -1343,7 +1343,7 @@ steps:
       from_secret: gpg_private_key
     GPG_PUBLIC_KEY:
       from_secret: gpg_public_key
-  image: grafana/agent-build-image:0.29.0
+  image: grafana/agent-build-image:0.30.0
   name: Publish release
   volumes:
   - name: docker
@@ -1368,7 +1368,7 @@ steps:
   - DOCKER_OPTS="" make dist/grafana-agentctl-linux-amd64
   - DOCKER_OPTS="" make dist.temp/grafana-agent-flow-linux-amd64
   - DOCKER_OPTS="" make test-packages
-  image: grafana/agent-build-image:0.29.0
+  image: grafana/agent-build-image:0.30.0
   name: Test Linux system packages
   volumes:
   - name: docker
@@ -1464,6 +1464,6 @@ kind: secret
 name: updater_private_key
 ---
 kind: signature
-hmac: 25d237e0e89cc8cb868363dcb1c0016776b16181321abf804d210f5762d79f5d
+hmac: 94dfb027f98cde4ce723862940c7d22765cbf23ed9ec4e69790d3ed124154618
 
 ...

--- a/.github/workflows/test.yml
+++ b/.github/workflows/test.yml
@@ -16,10 +16,10 @@ jobs:
     steps:
     - name: Checkout code
       uses: actions/checkout@v3
-    - name: Set up Go 1.20
+    - name: Set up Go 1.21
       uses: actions/setup-go@v3
       with:
-        go-version: "1.20.3"
+        go-version: "1.21.0"
         cache: true
     - name: Test
       run: make GO_TAGS="nodocker" test

--- a/CHANGELOG.md
+++ b/CHANGELOG.md
@@ -10,6 +10,10 @@ internal API changes are not present.
 Main (unreleased)
 -----------------
 
+### Other changes
+
+- Use Go 1.21.0 for builds. (@rfratto)
+
 v0.36.0-rc.0 (2023-08-25)
 --------------------
 
@@ -23,7 +27,7 @@ v0.36.0-rc.0 (2023-08-25)
   to enable decompression explicitly. See the [upgrade guide][] for migration
   instructions. (@thampiotr)
 
-- `otelcol.exporter.prometheus`: Set `include_scope_info` to `false` by default. You can set 
+- `otelcol.exporter.prometheus`: Set `include_scope_info` to `false` by default. You can set
   it to `true` to preserve previous behavior. (@gouthamve)
 
 - `prometheus.remote_write`: Set `retry_on_http_429` to `true` by default in the `queue_config` block.
@@ -81,7 +85,7 @@ v0.36.0-rc.0 (2023-08-25)
   - `discovery.hetzner` - service discovery for Hetzner Cloud. (@marctc)
   - `discovery.nomad` - service discovery from Nomad. (@captncraig)
   - `discovery.puppetdb` - service discovery from PuppetDB. (@captncraig)
-  - `otelcol.processor.discovery` adds resource attributes to spans, where the attributes 
+  - `otelcol.processor.discovery` adds resource attributes to spans, where the attributes
   keys and values are sourced from `discovery.*` components. (@ptodev)
   - `otelcol.connector.spanmetrics` - creates OpenTelemetry metrics from traces. (@ptodev)
 
@@ -112,7 +116,7 @@ v0.36.0-rc.0 (2023-08-25)
 - Fix an issue that lead the `loki.source.docker` container to use excessive
   CPU and memory. (@tpaschalis)
 
-- Fix issue where `otelcol.exporter.loki` was not normalizing label names 
+- Fix issue where `otelcol.exporter.loki` was not normalizing label names
   to comply with Prometheus conventions. (@ptodev)
 
 - Agent Management: Fix issue where an integration defined multiple times could lead to undefined behaviour. (@jcreixell)
@@ -137,7 +141,7 @@ v0.35.3 (2023-08-09)
 
 - (Agent static mode) Jaeger remote sampling works again, through a new `jaeger_remote_sampling`
   entry in the traces config. It is no longer configurable through the jaeger receiver.
-  Support Jaeger remote sampling was removed accidentally in v0.35, and it is now restored, 
+  Support Jaeger remote sampling was removed accidentally in v0.35, and it is now restored,
   albeit via a different config entry.
 
 - Clustering: Nodes take part in distributing load only after loading their
@@ -232,8 +236,8 @@ v0.35.0 (2023-07-18)
 
 - The `remote_sampling` block has been removed from `otelcol.receiver.jaeger`. (@ptodev)
 
-- (Agent static mode) Jaeger remote sampling used to be configured using the Jaeger receiver configuration. 
-  This receiver was updated to a new version, where support for remote sampling in the receiver was removed. 
+- (Agent static mode) Jaeger remote sampling used to be configured using the Jaeger receiver configuration.
+  This receiver was updated to a new version, where support for remote sampling in the receiver was removed.
   Jaeger remote sampling is available as a separate configuration field starting in v0.35.3. (@ptodev)
 
 ### Deprecations

--- a/cmd/grafana-agent-operator/Dockerfile
+++ b/cmd/grafana-agent-operator/Dockerfile
@@ -4,7 +4,7 @@
 # default when running `docker buildx build` or when DOCKER_BUILDKIT=1 is set
 # in environment variables.
 
-FROM --platform=$BUILDPLATFORM grafana/agent-build-image:0.29.0 as build
+FROM --platform=$BUILDPLATFORM grafana/agent-build-image:0.30.0 as build
 ARG BUILDPLATFORM
 ARG TARGETPLATFORM
 ARG TARGETOS

--- a/cmd/grafana-agent/Dockerfile
+++ b/cmd/grafana-agent/Dockerfile
@@ -4,7 +4,7 @@
 # default when running `docker buildx build` or when DOCKER_BUILDKIT=1 is set
 # in environment variables.
 
-FROM --platform=$BUILDPLATFORM grafana/agent-build-image:0.29.0 as build
+FROM --platform=$BUILDPLATFORM grafana/agent-build-image:0.30.0 as build
 ARG BUILDPLATFORM
 ARG TARGETPLATFORM
 ARG TARGETOS

--- a/cmd/grafana-agent/Dockerfile.windows
+++ b/cmd/grafana-agent/Dockerfile.windows
@@ -1,4 +1,4 @@
-FROM grafana/agent-build-image:0.29.0-windows as builder
+FROM grafana/agent-build-image:0.30.0-windows as builder
 ARG VERSION
 ARG RELEASE_BUILD=1
 

--- a/cmd/grafana-agentctl/Dockerfile
+++ b/cmd/grafana-agentctl/Dockerfile
@@ -4,7 +4,7 @@
 # default when running `docker buildx build` or when DOCKER_BUILDKIT=1 is set
 # in environment variables.
 
-FROM --platform=$BUILDPLATFORM grafana/agent-build-image:0.29.0 as build
+FROM --platform=$BUILDPLATFORM grafana/agent-build-image:0.30.0 as build
 ARG BUILDPLATFORM
 ARG TARGETPLATFORM
 ARG TARGETOS

--- a/cmd/grafana-agentctl/Dockerfile.windows
+++ b/cmd/grafana-agentctl/Dockerfile.windows
@@ -1,4 +1,4 @@
-FROM grafana/agent-build-image:0.29.0-windows as builder
+FROM grafana/agent-build-image:0.30.0-windows as builder
 ARG VERSION
 ARG RELEASE_BUILD=1
 

--- a/component/loki/source/syslog/internal/syslogtarget/syslogtarget_test.go
+++ b/component/loki/source/syslog/internal/syslogtarget/syslogtarget_test.go
@@ -760,7 +760,7 @@ func testSyslogTargetWithTLSVerifyClientCertificate(t *testing.T, fmtFunc format
 
 		buf := make([]byte, 1)
 		_, err = c.Read(buf)
-		require.EqualError(t, err, "remote error: tls: bad certificate")
+		require.ErrorContains(t, err, "remote error: tls:")
 	})
 
 	t.Run("WithClientCertificate", func(t *testing.T) {

--- a/converter/internal/staticconvert/testdata/integrations.river
+++ b/converter/internal/staticconvert/testdata/integrations.river
@@ -34,6 +34,28 @@ prometheus.scrape "integrations_blackbox" {
 	job_name   = "integrations/blackbox"
 }
 
+prometheus.exporter.snmp "integrations_snmp" {
+	target "network_switch_1" {
+		address     = "192.168.1.2"
+		module      = "if_mib"
+		auth        = "public"
+		walk_params = "public"
+	}
+
+	target "network_router_2" {
+		address     = "192.168.1.3"
+		module      = "mikrotik"
+		auth        = "private"
+		walk_params = "private"
+	}
+}
+
+prometheus.scrape "integrations_snmp" {
+	targets    = prometheus.exporter.snmp.integrations_snmp.targets
+	forward_to = [prometheus.remote_write.integrations.receiver]
+	job_name   = "integrations/snmp"
+}
+
 prometheus.exporter.cloudwatch "integrations_cloudwatch_exporter" {
 	sts_region    = "us-east-2"
 	fips_disabled = false
@@ -313,28 +335,6 @@ prometheus.scrape "integrations_redis_exporter" {
 	targets    = discovery.relabel.integrations_redis_exporter.output
 	forward_to = [prometheus.remote_write.integrations.receiver]
 	job_name   = "integrations/redis_exporter"
-}
-
-prometheus.exporter.snmp "integrations_snmp" {
-	target "network_switch_1" {
-		address     = "192.168.1.2"
-		module      = "if_mib"
-		auth        = "public"
-		walk_params = "public"
-	}
-
-	target "network_router_2" {
-		address     = "192.168.1.3"
-		module      = "mikrotik"
-		auth        = "private"
-		walk_params = "private"
-	}
-}
-
-prometheus.scrape "integrations_snmp" {
-	targets    = prometheus.exporter.snmp.integrations_snmp.targets
-	forward_to = [prometheus.remote_write.integrations.receiver]
-	job_name   = "integrations/snmp"
 }
 
 prometheus.exporter.snowflake "integrations_snowflake" {

--- a/go.mod
+++ b/go.mod
@@ -149,7 +149,7 @@ require (
 	github.com/prometheus/prometheus v1.99.0
 	github.com/prometheus/snmp_exporter v0.22.1-0.20230623130038-562ae9055ce3
 	github.com/prometheus/statsd_exporter v0.22.8
-	github.com/pyroscope-io/godeltaprof v0.1.1
+	github.com/pyroscope-io/godeltaprof v0.1.2
 	github.com/richardartoul/molecule v1.0.1-0.20221107223329-32cfee06a052
 	github.com/rs/cors v1.9.0
 	github.com/shirou/gopsutil/v3 v3.23.5
@@ -620,8 +620,6 @@ require (
 	sigs.k8s.io/json v0.0.0-20221116044647-bc3834ca7abd // indirect
 	sigs.k8s.io/structured-merge-diff/v4 v4.2.3 // indirect
 )
-
-require github.com/foxcpp/go-mockdns v1.0.0
 
 require (
 	github.com/drone/envsubst v1.0.3 // indirect

--- a/pkg/server/tls_test.go
+++ b/pkg/server/tls_test.go
@@ -64,5 +64,5 @@ func Test_tlsListener(t *testing.T) {
 
 	var urlError *url.Error
 	require.ErrorAs(t, err, &urlError)
-	require.Contains(t, urlError.Err.Error(), "tls: bad certificate")
+	require.Contains(t, urlError.Err.Error(), "tls:")
 }

--- a/tools/crow/Dockerfile
+++ b/tools/crow/Dockerfile
@@ -4,7 +4,7 @@
 # default when running `docker buildx build` or when DOCKER_BUILDKIT=1 is set
 # in environment variables.
 
-FROM --platform=$BUILDPLATFORM grafana/agent-build-image:0.29.0 as build
+FROM --platform=$BUILDPLATFORM grafana/agent-build-image:0.30.0 as build
 ARG BUILDPLATFORM
 ARG TARGETPLATFORM
 ARG TARGETOS

--- a/tools/make/build-container.mk
+++ b/tools/make/build-container.mk
@@ -34,7 +34,7 @@
 # variable names should be passed through to the container.
 
 USE_CONTAINER       ?= 0
-BUILD_IMAGE_VERSION ?= 0.29.0
+BUILD_IMAGE_VERSION ?= 0.30.0
 BUILD_IMAGE         ?= grafana/agent-build-image:$(BUILD_IMAGE_VERSION)
 DOCKER_OPTS         ?= -it
 

--- a/tools/smoke/Dockerfile
+++ b/tools/smoke/Dockerfile
@@ -4,7 +4,7 @@
 # default when running `docker buildx build` or when DOCKER_BUILDKIT=1 is set
 # in environment variables.
 
-FROM --platform=$BUILDPLATFORM grafana/agent-build-image:0.29.0 as build
+FROM --platform=$BUILDPLATFORM grafana/agent-build-image:0.30.0 as build
 ARG BUILDPLATFORM
 ARG TARGETPLATFORM
 ARG TARGETOS

